### PR TITLE
Issue 6493: Improving the CLI's usability through ENV variables

### LINKED ISSUE #6493: Improving the CLI's usability through ENV variables
**Is your feature request related to a problem? Please describe.**
When running the Admin CLI on pravega/SDP deployments, making sure that the Admin CLI's config file is filled with the required values can be tedious. Updating the config with the correct values can take longer than debugging the original issue it was being used for.

**Describe the solution you'd like**
This is the first step towards trying to auto fill the CLI's config with the correct values during deployment itself. Most of these valid values are available in the environment variables present on the controller and segmentstore pods. We need to add support for the CLI to recognize and obtain values from the environment variables.

### PR DESCRIPTION
**Pull request by anirudhkovuru**
**Change log description**  
- Added environment variable support to the Admin CLI config.
- The variable name should be provided with `$` as the prefix. For example:
    - If the variable is defined as `export ADMIN_CLI_VAR=localhost:9090`. The value should be provided in the config as `$ADMIN_CLI_VAR`.
    - In the `admin-cli.properties` the value would be provided as `cli.controller.connect.grpc.uri=$ADMIN_CLI_VAR`.
- The `config set` command also allows for environment variables to be provided.

**Purpose of the change**  
Fixes #6493, #6504.

**What the code does**  
- Added a new method `getIfEnv` in `ConfigUtils` that checks if the value provided starts with the prefix `$`. If it does, then the environment variable's value is obtained.
- This method is used in the `ConfigSetCommand` and during the loading of the properties file.

**How to verify it**  
All tests must pass.

### CODE REVIEW

**Comment by RaulGracia** [comment]
@alfred-landrum This PR is supposed to make easier the initial configuration of the Admin CLI when run inside any of the Pravega pods. Maybe you can provide feedback here based on your experience.

**Comment by RaulGracia** [comment]
@anirudhkovuru maybe this PR can also fix the discrepancies in configuring `pravegaservice.clusterName`. In Pravega, we assume that the cluster name will always start by `pravega`, so whatever we put in there will be added as a suffix (`pravega/xxx`). However, the CLI is not doing that assumption, so it expects the full `pravega/xxx` to be set. This is causing trouble when configuring the CLI and maybe it can be considered to be fixed here as well, what do you think?

**Review by pbelgundi — CHANGES_REQUESTED** [review]
@anirudhkovuru It may help to state in the PR description, the names of cli-properties and corresponding Environment variable names we plan to default them to.

File: cli/admin/src/main/java/io/pravega/cli/admin/AdminCommandState.java [hunk 1]
```diff
@@ -47,7 +51,14 @@
     public AdminCommandState() throws IOException {
         this.configBuilder = ServiceBuilderConfig.builder();
         try {
-            this.configBuilder.include(System.getProperty(ServiceBuilderConfig.CONFIG_FILE_PROPERTY_NAME, "conf/admin-cli.properties"));
+            Properties properties = new Properties();
+            try (FileReader reader = new FileReader(System.getProperty(ServiceBuilderConfig.CONFIG_FILE_PROPERTY_NAME, "conf/admin-cli.properties"))) {
```
Comment (pbelgundi) [inline cli/admin/src/main/java/io/pravega/cli/admin/AdminCommandState.java:55]: Is `admin-cli.properties` a generated file? I don't see any changes to this file that replaces the existing values with Env variable names.

**Review by pbelgundi — CHANGES_REQUESTED** [review]
Based on the meeting minutes, I think we decided that the default value for `tls.enabled` should be `false`. I could not find that change in this PR.

**Review by pbelgundi — CHANGES_REQUESTED** [review]
Does just moving to using environment variables solve the problem of commands not working on CLI in certain environments?

**Review by pbelgundi — COMMENTED** [review]
Since `auth` is enabled by default, default values for `cli.credentials.username` and `cli.credentials.pwd` should point to environment variable names used by the operator to exposing the auth secret.
This may require changes on operator as well. Or are they already in place?

**Comment by anirudhkovuru** [comment]
> Based on the meeting minutes, I think we decided that the default value for `tls.enabled` should be `false`. I could not find that change in this PR.

It has been made `false`.

File: cli/admin/src/main/java/io/pravega/cli/admin/AdminCommandState.java (hunk 1 above)
Comment (anirudhkovuru) [reply cli/admin/src/main/java/io/pravega/cli/admin/AdminCommandState.java:55]: This will be handled in #6506.

**Comment by anirudhkovuru** [comment]
> @anirudhkovuru maybe this PR can also fix the discrepancies in configuring `pravegaservice.clusterName`. In Pravega, we assume that the cluster name will always start by `pravega`, so whatever we put in there will be added as a suffix (`pravega/xxx`). However, the CLI is not doing that assumption, so it expects the full `pravega/xxx` to be set. This is causing trouble when configuring the CLI and maybe it can be considered to be fixed here as well, what do you think?

I've made this into another issue #6505. This change can be provided there.

**Comment by anirudhkovuru** [comment]
> Since `auth` is enabled by default, default values for `cli.credentials.username` and `cli.credentials.pwd` should point to environment variable names used by the operator to exposing the auth secret. This may require changes on operator as well. Or are they already in place?

Yes this requires an operator change to make the `auth` related values accessible through ENV variables.

**Comment by anirudhkovuru** [comment]
> @anirudhkovuru It may help to state in the PR description, the names of cli-properties and corresponding Environment variable names we plan to default them to.

A list of corresponding environment variables can be found [here](https://github.com/pravega/pravega/issues/6506#issuecomment-1007298354).

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments:
- Make sure that we cover properly the environment variable loading logic.
- You may want to include the `clusterName` logic fix in this PR (tiny addition).
- Add a method to cleanup endpoint values in which the user mistakenly added the prefix (like `http://` or `tls://`). The CLI already sets this, so we can clean these up from input config.

**Comment by anirudhkovuru** [comment]
> Looks good, just a couple of comments:
> 
> * Make sure that we cover properly the environment variable loading logic.
> * You may want to include the `clusterName` logic fix in this PR (tiny addition).
> * Add a method to cleanup endpoint values in which the user mistakenly added the prefix (like `http://` or `tls://`). The CLI already sets this, so we can clean these up from input config.

I have added a cleanup for the endpoint values.
The `clusterName` fix is being taken care of by @tewit [here](https://github.com/pravega/pravega/pull/6525)

File: cli/admin/src/main/java/io/pravega/cli/admin/utils/ConfigUtils.java [hunk 2]
```diff
@@ -46,9 +49,20 @@ public static void loadProperties(AdminCommandState state) {
...
         state.getConfigBuilder().include(pravegaProperties);
     }
+
+    public static String getIfEnv(String value) {
+        if (value.startsWith("$")) {
+            String envValue = System.getenv(value.substring(1));
+            if (envValue == null) {
+                throw new IllegalArgumentException(String.format("%s is not a valid environment variable.", value));
+            }
+            return envValue;
```
[the remark this answers is not in the document]
Comment (anirudhkovuru) [reply cli/admin/src/main/java/io/pravega/cli/admin/utils/ConfigUtils.java:64]: I've added a test for this.

**Review by pbelgundi — APPROVED** [review]
Assuming the changes in this PR are restricted to enabling use of environment variables in the `admin-cli.properties` file, and other changes will be handled by different PRs, this LGTM.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**